src/simsopt/field/magneticfieldclasses.py

```diff
@@ -502,6 +503,179 @@ def from_dict(cls, d, serial_objs_dict, recon_objs):
         return field
 
 
+class DipoleField(MagneticField):
+    r"""
+    Computes the MagneticField induced by N dipoles. This is very simple but needs to be
+    a type MagneticField class for using the other simsopt functionality.
+    The field is given by
+
+    .. math::
+
+        B(\mathbf{x}) = \frac{\mu_0}{4\pi} \sum_{i=1}^{N} (\frac{3\mathbf{r}_i\cdot \mathbf{m}_i}{|\mathbf{r}_i|^5}\mathbf{r}_i - \frac{\mathbf{m}_i}{|\mathbf{r}_i|^3})
+
+    where :math:`\mu_0=4\pi 10^{-7}` is the magnetic constant and :math:\mathbf{r_i} = \mathbf{x} - \mathbf{x}^{dipole}_i is the vector between the field evaluation point and the dipole i position.
+
+    Args:
+        pm_opt: A PermanentMagnetOptimizer object that has already been optimized.
+    """
+
+    def __init__(self, pm_opt=None):
```

Comment:
You are creating a depening on PermanentMagnetOptimizer class for DipoleField class? Is this really necessary? You are limiting the usage of DipoleField.

Reply:
Agreed, this was unnecessary and went through and fixed this